src/main/java/com/healthapp/project/DAO/DataSourceConfig.java

```diff
@@ -1,31 +1,23 @@
 package com.healthapp.project.DAO;
 
-import org.springframework.beans.factory.annotation.Value;
+import javax.sql.DataSource;
+
+import com.healthapp.project.Service.UserService;
+import org.postgresql.ds.PGSimpleDataSource;
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
 import org.springframework.jdbc.datasource.DriverManagerDataSource;
 
-import javax.sql.DataSource;
-
 @Configuration
 public class DataSourceConfig {
 
-    @Value("${DB_URL_TEST:jdbc:postgres://dex:JdjDKIMLKfDoiGyL4qn4wsuOZ7n6ByjP@dpg-cg1446o2qv25u2njfrk0-a.ohio-postgres.render.com/empty_db}")
-    private String url;
-
-    @Value("${DB_USERNAME_TEST:dex}")
-    private String username;
-
-    @Value("${DB_PASSWORD_TEST:JdjDKIMLKfDoiGyL4qn4wsuOZ7n6ByjP}")
-    private String password;
+private final String username = System.getenv("DB_USERNAME_TEST");
 
-    @Bean
-    public DataSource dataSource(){
-        DriverManagerDataSource dataSource = new DriverManagerDataSource();
-        dataSource.setDriverClassName("org.postgresql.Driver");
-        dataSource.setUrl(url);
-        dataSource.setUsername(username);
-        dataSource.setPassword(password);
-        return dataSource;
-    }
+private final String password = System.getenv("DB_PASSWORD_TEST");
+  @Bean
+  public DataSource dataSource() {
+  DriverManagerDataSource dataSource = new DriverManagerDataSource();
+    dataSource.setUrl("jdbc:postgresql://dpg-cg1446o2qv25u2njfrk0-a.ohio-postgres.render.com:5432/empty_db?user=" + username +"&password=" + password +"");
+    return dataSource;
+  }
```

Comment:
A bit confusing here since i've been fiddling with diff techniques. but basically what i ended up doing was only setting the url field in the Datasource object to the entire url, containing the password and the username. Instead of just building the pieces separately if that makes sense. I was able to hide away the username and password into environment variables so you'll have to add them to your IDE once you pull this version for the connection to work. In a later version of this i'll probably do the same to the endpoint string here and maybe use a String formatter to build it instead.

Alsoooooo... this datasource object will be injected into the UserService as it'll need it as a dependency.  I also annotated the object as a bean so spring knows to come pick it up when it's needed